### ISSUE TITLE
XCSoar not saving igc file since version 7.xx

### ISSUE THREAD

**Issue by andymel2** [XCSoar/XCSoar#782]
XCSoar versions having and not having the problem
-------------------------------------------------

Which XCSoar version are you using when seeing the problem?  
7.19-PC, 7.20-PC
7.4 WIN64
7.2 WIN64
7.1 WIN64

Which XCSoar version is the last one you know that did not show the problem? 
6.8.17-PC

Expected behavior
-----------------
At the end of a task (in my case, virtual gliding flight) XCSoar should save an .igc file

Actual behavior
---------------
For version 6.xx, the file was saved correctly, in the XCSoarDat/Logs folder.  
For version 7.xx, I can see the igc file in the folder, while the program is running (while the flight is in progress), but as soon as XCSoar shuts down, the file disappears.

Steps to reproduce the behavior
-------------------------------

To reproduce the error, load a task (.tsk) file; run the simulation (fly the task).  While the flight/task is in progress, monitor the XCSoarData/Logs folder - the igc file is visible.
At the end of the flight/task, close XCSoar - file disappears!

Do you have any idea what may have caused this?
-----------------------------------------------

This might give us ideas we haven't thought of, feel free to suggest causes. NO,  It worked in 6.xx (and still does); stopped working in 7.xx

Do you have an idea how to solve the issue?
-------------------------------------------

**Comment by andymel2** [comment]
Would be very grateful if version 7.xx would save the igc file!

**Comment by andymel2** [comment]
Did not mean to close it!

**Comment by lordfolken** [comment]
There isn't even the most basic information in this bug report. No one can help you like this.

**Comment by andymel2** [comment]
This was not an empty report when I wrote it!  All the above sections were completed.

However, it is quite a basic problem.  For XCSoar 6.xx (PC Version), the .igc file was automatically saved at the completion of a task; it was to be found in XCSoarData/Logs and was still there after shutting down XCSoar.

Since I started using 7.xx, I can see the .igc file in the logs folder while the task is running, but as soon as I end the task and shut down XCSoar, the .igc file disappears from the logs folder.  I have searched my PC to see if it is being moved elsewhere - no.

I only use the PC version of XCSoar.

The igc file is, of course desirable for flight analysis.

**Comment by lordfolken** [comment]
Ok, i could reproduce this with the WIN64 target with the current HEAD.
The file is indeed deleted as soon as XCSoar exits

**Comment by lordfolken** [comment]
```
[2021-11-18T17:27:06Z] Starting XCSoar X
[2021-11-18T17:27:06Z] Display dpi=96,96
[2021-11-18T17:27:06Z] Initialise fonts
[2021-11-18T17:27:07Z] Startup dialog
[2021-11-18T17:27:07Z] Loading profiles
[2021-11-18T17:27:07Z] Loaded profile from C
[2021-11-18T17:27:07Z] Units: GetUserDefaultUILanguage() = 0x809
[2021-11-18T17:27:07Z] Loading language file
[2021-11-18T17:27:07Z] Language: GetUserDefaultUILanguage()=0x809
[2021-11-18T17:27:07Z] Loading input events file
[2021-11-18T17:27:07Z] LoadTerrain
[2021-11-18T17:27:07Z] Loading Topography File...
[2021-11-18T17:27:07Z] ReadWaypoints
[2021-11-18T17:27:07Z] Failed to read waypoint file: C
[2021-11-18T17:27:07Z] Failed to read waypoint file: waypoints.xcw
[2021-11-18T17:27:07Z] Failed to open 'airfields.txt' from ZIP file
[2021-11-18T17:27:07Z] RASP load
[2021-11-18T17:27:07Z] No rasp data file: Failed to open ZIP archive C:\users\folken\My Documents\XCSoarData\xcsoar-rasp.dat
[2021-11-18T17:27:07Z] ReadAirspace
[2021-11-18T17:27:07Z] Failed to load airspaces from map file: Failed to open 'airspace.txt' from ZIP file
[2021-11-18T17:27:07Z] Register serial devices
[2021-11-18T17:27:08Z] ProgramStarted
[2021-11-18T17:27:08Z] cannot open C:\users\folken\My Documents\XCSoarData\lua\init.lua: No such file or directory
[2021-11-18T17:27:08Z] Start at terrain center
[2021-11-18T17:27:08Z] Opening device T
[2021-11-18T17:27:39Z] Logger Started: C
[2021-11-18T17:27:52Z] Entering shutdown...
[2021-11-18T17:27:52Z] Logger stopped: C
[2021-11-18T17:27:52Z] Stop threads
[2021-11-18T17:27:52Z] Waiting for calculation thread
[2021-11-18T17:27:52Z] Waiting for draw thread
[2021-11-18T17:27:52Z] delete MapWindow
[2021-11-18T17:27:52Z] Save default task
[2021-11-18T17:27:52Z] Stop COM devices
[2021-11-18T17:27:52Z] Close Progress Dialog
[2021-11-18T17:27:52Z] Close Windows - main
[2021-11-18T17:27:52Z] Finished shutdown
```

**Comment by lordfolken** [comment]
The missing ':' after the c: is suspicous.

**Comment by lordfolken** [comment]
So this cost me 3 evenings.. I learnt quite a bit about gdb, strace and c++. 
In the end I found the function that deletes the file. It was duplicated between windows and the rest of the platforms. 
Some logic with exceptions was added over time, that is why it worked on the Linux side. the Logic never made it to the windows variant of the function, which is why the file was always deleted without exception. 

My proposal merges the two functions and only uses the platform specific deletion call.
